### ISSUE TITLE
Pax Dei (1995520)

### ISSUE THREAD

**Issue by ca1f** [ValveSoftware/Proton#7846]
# Compatibility Report
- Name of the game with compatibility issues: Pax Dei
- Steam AppID of the game: 1995520

## System Information
- GPU: AMD Radeon RX 6900 XT
- Video driver version: Mesa 24.1.1
- Kernel version: 6.9.5-arch1-1
- Link to full system information report as [Gist](https://gist.github.com/ca1f/f79ee6188087cfe6966e5c6498347377):
- Proton version: Proton Experimental

## I confirm:
- [x] that I have checked whether there are updates for my system available.
- [x] that I haven't found an existing compatibility report for this game.

[steam-1995520.log.gz](https://github.com/user-attachments/files/15892065/steam-1995520.log.gz)

## Symptoms 

* FPS sporadically drops to 0 for some time while walking in the world, sometimes the client recovers in time, sometimes timing out and getting kicked by the game server.
* Unable to open the map, FPS drops to 0 and client lags fully until being kicked from the server.
* Loading takes an unusual large amount of time.

## Reproduction

Always.

Edited 19-06-2024:
Added missing information and fixed some spelling mistakes.

**Comment by hsmalley** [comment]
I'm seeing the same thing. When I look at the game log I'm seeing a disconnect/timeouts every time I get dropped back into the menu. My friends running windows haven't had this problem. But I've seen other people in their discord saying they're having the same problems. Not sure what side the problem is on

```
[2024.06.19-22.05.42:704][994]LogNet: Warning: UNetConnection::Tick: Connection TIMED OUT. Closing connection.. Elapsed: 0.02, Real: 56.45, Good: 56.45, DriverTime: 37.52, Threshold: 15.00, [UNetConnection] RemoteAddr: 54.80.183.23:31314, Name: PDNetConnection_2147468527, Driver: GameNetDriver PDNetDriver_2147468528, IsServer: NO, PC: Controller_23113924-17fa-4772-a81f-68821e4e16b6, Owner: Controller_23113924-17fa-4772-a81f-68821e4e16b6, UniqueId: STEAM:STEAMID_REDACTED
[2024.06.19-22.05.42:705][994]LogNet: Error: UEngine::BroadcastNetworkFailure: FailureType = ConnectionTimeout, ErrorString = UNetConnection::Tick: Connection TIMED OUT. Closing connection.. Elapsed: 0.02, Real: 56.45, Good: 56.45, DriverTime: 37.52, Threshold: 15.00, [UNetConnection] RemoteAddr: 54.80.183.23:31314, Name: PDNetConnection_2147468527, Driver: GameNetDriver PDNetDriver_2147468528, IsServer: NO, PC: Controller_23113924-17fa-4772-a81f-68821e4e16b6, Owner: Controller_23113924-17fa-4772-a81f-68821e4e16b6, UniqueId: STEAM:STEAMID_REDACTED, Driver = GameNetDriver PDNetDriver_2147468528
[2024.06.19-22.05.42:705][994]LogPaxDei: Error: Network Failure: ConnectionTimeout
[2024.06.19-22.05.42:705][994]LogBlueprintUserMessages: [BP_NetworkingSystem_C] OnNetworkFailure: ConnectionTimeout
[2024.06.19-22.05.42:705][994]LogNet: Warning: Network Failure: GameNetDriver[ConnectionTimeout]: UNetConnection::Tick: Connection TIMED OUT. Closing connection.. Elapsed: 0.02, Real: 56.45, Good: 56.45, DriverTime: 37.52, Threshold: 15.00, [UNetConnection] RemoteAddr: 54.80.183.23:31314, Name: PDNetConnection_2147468527, Driver: GameNetDriver PDNetDriver_2147468528, IsServer: NO, PC: Controller_23113924-17fa-4772-a81f-68821e4e16b6, Owner: Controller_23113924-17fa-4772-a81f-68821e4e16b6, UniqueId: STEAM:STEAMID_REDACTED
[2024.06.19-22.05.42:705][994]LogPaxDei: Error: OnNetworkFailure: 'UNetConnection::Tick: Connection TIMED OUT. Closing connection.. Elapsed: 0.02, Real: 56.45, Good: 56.45, DriverTime: 37.52, Threshold: 15.00, [UNetConnection] RemoteAddr: 54.80.183.23:31314, Name: PDNetConnection_2147468527, Driver: GameNetDriver PDNetDriver_2147468528, IsServer: NO, PC: Controller_23113924-17fa-4772-a81f-68821e4e16b6, Owner: Controller_23113924-17fa-4772-a81f-68821e4e16b6, UniqueId: STEAM:STEAMID_REDACTED'
[2024.06.19-22.05.42:705][994]LogNet: NetworkFailure: ConnectionTimeout, Error: 'UNetConnection::Tick: Connection TIMED OUT. Closing connection.. Elapsed: 0.02, Real: 56.45, Good: 56.45, DriverTime: 37.52, Threshold: 15.00, [UNetConnection] RemoteAddr: 54.80.183.23:31314, Name: PDNetConnection_2147468527, Driver: GameNetDriver PDNetDriver_2147468528, IsServer: NO, PC: Controller_23113924-17fa-4772-a81f-68821e4e16b6, Owner: Controller_23113924-17fa-4772-a81f-68821e4e16b6, UniqueId: STEAM:STEAMID_REDACTED'
[2024.06.19-22.05.42:705][994]LogNet: UNetConnection::Close: [UNetConnection] RemoteAddr: 54.80.183.23:31314, Name: PDNetConnection_2147468527, Driver: GameNetDriver PDNetDriver_2147468528, IsServer: NO, PC: Controller_23113924-17fa-4772-a81f-68821e4e16b6, Owner: Controller_23113924-17fa-4772-a81f-68821e4e16b6, UniqueId: STEAM:STEAMID_REDACTED, Channels: 91, Time: 2024.06.19-22.05.42
[2024.06.19-22.05.42:705][994]LogNet: UNetConnection::SendCloseReason:
[2024.06.19-22.05.42:705][994]LogNet:  - Result=ConnectionTimeout, ErrorContext="ConnectionTimeout"
[2024.06.19-22.05.42:705][994]LogNet: UChannel::Close: Sending CloseBunch. ChIndex == 0. Name: [UChannel] ChIndex: 0, Closing: 0 [UNetConnection] RemoteAddr: 54.80.183.23:31314, Name: PDNetConnection_2147468527, Driver: GameNetDriver PDNetDriver_2147468528, IsServer: NO, PC: Controller_23113924-17fa-4772-a81f-68821e4e16b6, Owner: Controller_23113924-17fa-4772-a81f-68821e4e16b6, UniqueId: STEAM:STEAMID_REDACTED
[2024.06.19-22.06.12:164][995]LogNet: Connection failed; returning to Entry
```

**Comment by ahjolinna** [comment]
the game crashes xwayland at times, my guess because the load becomes to heavy, I'm getting this error:
`[ 2215.647500] [ T3043] [drm:nv_drm_gem_alloc_nvkms_memory_ioctl [nvidia_drm]] ERROR [nvidia-drm] [GPU ID 0x00000100] Failed to allocate NVKMS memory for GEM object `
need to reboot the system every time.

not sure if this is nvidia-driver issue or xwayland or game or proton.


here is a 43mb log file using  proton-ge : [download link](https://www.swisstransfer.com/d/093dc33c-14cc-4e81-a05a-6ed7a46f5533)

-----

my spec:

```
Operating System: openSUSE Tumbleweed 20240618
KDE Plasma Version: 6.1.80
KDE Frameworks Version: 6.4.0
Qt Version: 6.7.1
Kernel Version: 6.9.4-1-default (64-bit)
Graphics Platform: Wayland
Processors: 16 × AMD Ryzen 7 5700G with Radeon Graphics
Memory: 46.3 GiB of RAM
Graphics Processor: NVIDIA GeForce RTX 3060 Ti/PCIe/SSE2 (driver v555.52.04)
```

**Comment by ahjolinna** [comment]
> ## Symptoms
> 
>     * FPS sporadically drops to 0 for some time while walking in the world, sometimes the client recovers in time, sometimes timing out and getting kicked by the game server.
> 
>     * Unable to open the map, FPS drops to 0 and client lags fully until being kicked from the server.
> 
>     * Loading takes an unusual large amount of time.


these mentioned issues are more or less game or server related issues and the devs are aware of them, they are trying to fix them asp (hopefully mostly resolved in a week)

I recommend follow their discord channel for more up-to-date info ; https://discord.gg/paxdei

here is an unofficial game issue tracker (made by community manager): https://trello.com/b/HKnGk1JW/pax-dei-known-issues

**Comment by hsmalley** [comment]
I suspect this has to do with proton / easy anti-cheat more than than the servers. On Windows there's no excessive load times, fps drops, map drops, etc. Getting kicked/dropped when opening the map smells like EAC to me TBH.

The performance & load times seem to be slightly better using Proton-GE too. I'm not seeing anything in the logs that would confirm that and without the hard data to go on it's hard to know exactly what is at fault. If someone has an idea on how to confirm it I'm happy to try it out. 

~They did post in discord that they're looking into the login problems still. Nothing mentioned about the other problems we're seeing. So maybe whatever they're doing will help us, but I'm doubtful.~

They posted that they login problems were fixed. I can confirm that whatever they did just made the problem worse, at least for me. I have to try to enter the world at least 5x for it to load. Then it's just a roll of the dice if/when I loose connection and get dropped into the lobby

**Comment by VirtualJunky** [comment]
Same issues here, extreme load times, map kicks you out to character select, and random freezes when exploring. Got about 20 hours so far playing with a group of 4 (3 Windows users and myself) and none of them have any of these issues, one has much worse specs then me.

For them its about 30-45 seconds from boot to be online and playing as Windows users. I can't even get past the broken splash screen in that time, it takes me about 5 minutes to get in. Once you actually get to character select and try to enter the world if the loading screen stalls to long (which is SUPER often) the server will disconnect you. With System Monitor open you can see the point where the network up/down goes to zero, and 10-15 seconds later you are back on character select. Likewise in-game and exploring sometimes you will get a long freeze, if its long enough you will see the same.

I've tried multiple versions of Proton, tried with and without multiple launch commands, went in and manually edited the graphics settings file and made every option zero (game now looks like its from the early 90s) and made absolutely zero difference.

I pretty sure these issues have nothing to do with server issues as until you get to character select you aren't actually connected to the server as shown by network activity, and the loading and freezing issues are still present. I can also confirm this has nothing to do with hardware as someone in my group has a 980 TI, and I have a 3080, they get in under a minute and have low-ish FPS as expected but don't have random drops to character select or 5-10 second long freezes, also their map works. All absolutely issues with some sort of conflict between Game/EAC/Proton.

My neofetch:
![Screenshot from 2024-06-24 03-53-36](https://github.com/ValveSoftware/Proton/assets/58893241/f68b0f10-3df1-411d-8455-23fd62cbdb9f)

**Comment by ca1f** [comment]
@VirtualJunky yep, this reflects exactly the same experience I have with the game. Insane loading times, and none of the people in my group who are playing on windows have a similar problem.

@hsmalley how did you acquire these debug logs?

**Comment by hsmalley** [comment]
@ca1f Check the log files for the game. 

I normally do 
`tail -f ~/.steam/steam/steamapps/compatdata/1995520/pfx/drive_c/users/steamuser/AppData/Local/PaxDei/Saved/Logs/PaxDei*`

so I can watch everything go by as I try to login.

**Comment by VirtualJunky** [comment]
Got some info I just found that I wanted to add. When running around the world my game often freezes for 1-3 seconds at a time, sometimes this will disconnect you, mentioned that above. But when I go into free cam (F7, and Alt+Scroll to adjust speed for those that don't know) I can zoom around the map with flawless performance. The only thing you don't see in free cam is animals and players, which makes me think the stuttering issue has to do with animals/players loading or unloading as you explore on foot.

Was curious if anyone else has the same results.

**Comment by hsmalley** [comment]
I opened a support ticket with them directly, [#6411](https://support.playpaxdei.com/hc/en-us/requests/6411) if you want to reference if/when you make your own ticket with them.

**Comment by olekw** [comment]
I submitted my own Support Ticket with Mainframe (6736) and referenced @hsmalley's as well.

I went through my Proton error log and I match OP on the first 20 but then I have the following (4 each):
2660.815:01c4:01c8:warn:seh:OutputDebugStringW L"Fatal error: [File:D:\\tc_work\\f667cdf70eb90d0e\\UE\\Engine\\Source\\Runtime\\RenderCore\\Private\\RenderingThread.cpp] [Line: 1274] \nGameThread timed out waiting for RenderThread after 120.00 secs\n"
2660.833:01c4:0304:warn:seh:OutputDebugStringW L"[2024.06.28-19.40.43:930][  6]LogWindows: Error: appError called: Fatal error: [File:D:\\tc_work\\f667cdf70eb90d0e\\UE\\Engine\\Source\\Runtime\\RenderCore\\Private\\RenderingThread.cpp] [Line: 1274] \nGameThread timed out waiting for RenderThread after 120.00 secs\n\r\n\r\n\r\n"
2660.835:01c4:0304:warn:seh:OutputDebugStringW L"[2024.06.28-19.40.43:932][  6]LogWindows: Windows GetLastError: Success. (0)\r\n"

At that point the game hangs and generally crashes if left on its own for an hour or so. Or locks up my whole computer...

GPU is RX 6900 XT and I was able to play fine in the first Pax Dei alpha but not anymore. Something Proton-related clearly changed!

**Comment by olekw** [comment]
> @hsmalley how did you acquire these debug logs?

You can also use this launch option in Steam to get a nice Proton error log conveniently deposited in your home directory:
`PROTON_LOG=1 %command%`

**Comment by hsmalley** [comment]
They responded to my ticket and asked for some screen captures. I sent them some recordings. They seem interested in hearing feedback and exploring a fix, would be pretty awesome of them if they do but not holding my breath

Also @olekw THANKS!!! I'll give that a go!!

**Comment by hsmalley** [comment]
They closed my ticket with the normal boiler plate of it's only supported on windows 10/11 etc etc... Sorry not sorry. Sad but expected :shrug:

**Comment by olekw** [comment]
I'm not sure how much of this is a Proton issue and how much of this is a developer-unwilling-to-make-any-effort-for-Proton-support issue. I'm opening another ticket to ask for a timeline for Proton support. If they can't provide at least an estimate, I'm going to ask for my money back. Hate doing that but, in the long run, I don't want to get involved with a company that doesn't care whether or not I can run their game.

**Comment by olekw** [comment]
Additional data point: the Pax Dei update on September 5th finally allowed me to start the game! But only with GE Proton 9-12. Proton Experimental still crashes on boot. Perhaps that will help someone understand the issue better.

**Comment by olekw** [comment]
Ok, today's Proton Experimental update works now! Well, at least it doesn't crash on boot anymore.

**Comment by Ozwald1** [comment]
I thing 2/3 of symptoms are now (mostly) over since last update 11/5, but that third one, long loading times, got way, way worse since it takes now ages to get in the game menu. Someone says +5minutes with very powerful PC and myself clocked yesterday exact 30 minutes(!!!) of black screen.  The game itself runs pretty fine when start with proton experimental now.

**Comment by hsmalley** [comment]
Can confirm! 30+min loadtime but it runs well once it has been loaded

**Comment by BroltzU** [comment]
I did some digging and found that font loading is adding significant load times. Renaming /usr/share/fonts before launching the game brought the loading time to just under 3 minutes whereas without this renaming the loading would take a bit over 9 minutes. On same system with comparable NVMe SSD (previous model from same manufacturer) and Windows, the loading takes 37 seconds from Launch to character select screen.

edit: If you have already run the game with your system fonts loaded, you need to clear out the `[Software\\Wine\\Fonts\\External Fonts]` section in your `~/.steam/steam/steamapps/compatdata/1995520/pfx/user.reg` (or similar).

**Comment by daela** [comment]
@BroltzU That is very encouraging news.  Looking forward to giving the font rename a try in a bit.

Outside of the long initial load time, the game is very playable for me.  The only thing I seem to run into is after some amount of time I get this camera stutter. Not sure how to describe it. If i'm running in a straight line everything is smooth but if I move the camera around the background/terrain/horizon etc.. stutters/is jerky.  I also get the same thing most of the time after I cross a region zone line.   Once it occurs, it doesn't go away until a close of client and relaunch. Going back to character select and back into the game doesn't resolve it.  Flames/smoke animations are smooth and look fine, it's only the terrain when rotating the camera.  Anyone else having similar issues?   

The stutter doesn't seem to be caused by flowerpots, particular areas or anything I can pin down. I can stand at a crafting table making something for 30 minutes, no issue. A bit later, I can stand in the same spot doing the same thing and maybe after 10 minutes the stutter starts. 

Due to the very long load times, I tend to just deal with the stuttering versus relaunching the client. If the font thing seems to cut down on load times, re-launching will become a reasonable workaround.

**Comment by BroltzU** [comment]
@daela 
> Replying to https://github.com/ValveSoftware/Proton/issues/7846#issuecomment-2493970463

I've experienced the same stuttering, usually starting some 30-60 minutes into a play session, and I think it might be related to input handling. You can use the auto-run, and as long as you don't move mouse or press any other keys, the game runs fine. But any mouse movement or pressing and holding of keys will cause bad stuttering.

edit: It would seem like activating the Steam Overlay for the game seems to somehow prevent the stuttering from occurring.

**Comment by simifor** [comment]
@BroltzU @daela regarding the slowdowns after some play time, adding the following to the game launch parameters might help avoid it: `LD_PRELOAD="" %command%`

**Comment by daela** [comment]
@simifor huzzah!  Can confirm that the `LD_PRELOAD=""` parameter does seem to take care of the slowdowns/stuttering.

**Comment by Ozwald1** [comment]
Wow!! With latest experimental starting time dropped 30+ minutes --> couple of seconds!  But, the game itself runs now less smoothly. Boderless mode freezes badly, but fullscreen seems to be playable.

**Comment by simifor** [comment]
I briefly tested the game on base experimental as well as bleeding edge but
borderless seemed to behave the same as fullscreen for me. Does the issue
appear from the start or only after playing for a while? You should be able
to use base experimental (launch times won't be affected), do you see the
problem there as well?

It would also help to have more information about your system (cpu/gpu,
kernel and driver version, etc).

**Comment by ahjolinna** [comment]
I just tested the game after ½ a year, and while it finally launches quickly/normally now, the overall performance has significantly regressed. I can't even get decent FPS anymore. Even at 1080p with DLSS (on or off) and low settings, the game struggles to stay above 30 FPS. Previously, I could easily get around 60 FPS, even at 4K (albeit on low settings or with DLSS).

What’s even more frustrating is that the performance seems equally bad regardless of the graphical settings I use.

------

```

                  -`                     ahjolinna@archlinux
                 .o+`                    -------------------
                `ooo/                    OS: Arch Linux x86_64
               `+oooo:                   Kernel: Linux 6.12.8-zen1-1-zen
              `+oooooo:                  Uptime: 24 mins
              -+oooooo+:                 Packages: 1034 (pacman), 35 (flatpak-system), 7 (flatpak-user)
            `/:-:++oooo+:                Shell: bash 5.2.37
           `/++++/+++++++:               Display (H277HK): 3840x2160 @ 60 Hz (as 3072x1728) in 27" [External, HDR]
          `/++++++++++++++:              DE: KDE Plasma 6.2.5
         `/+++ooooooooooooo/`            WM: KWin (Wayland)
        ./ooosssso++osssssso+`           WM Theme: Breeze
       .oossssso-````/ossssss+`          Theme: Breeze (Dark) [Qt], Default [GTK2/3/4]
      -osssssso.      :ssssssso.         Icons: breeze-dark [Qt], breeze-dark [GTK2/3/4]
     :osssssss/        osssso+++.        Font: IBM Plex Sans (10pt) [Qt], IBM Plex Sans (10pt) [GTK2/3/4]
    /ossssssss/        +ssssooo/-        Cursor: breeze (24px)
  `/ossssso+/:-        -:/+osssso+-      Terminal: yakuake 24.12.0
 `+sso+:-`                 `.-/+oso:     CPU: AMD Ryzen 7 5700G (16) @ 5.29 GHz
`++:.                           `-/+/    GPU: NVIDIA GeForce RTX 3060 Ti [Discrete] + "nvidia-open" driver
.`                                 `/    Memory: 7.63 GiB / 46.84 GiB (16%)
                                         Swap: 0 B / 4.00 GiB (0%)
                                         Disk (/): 409.73 GiB / 1.35 TiB (30%) - btrfs
```

**Comment by Ozwald1** [comment]
> Replying to https://github.com/ValveSoftware/Proton/issues/7846#issuecomment-2576136893

Looks like it was something in my system, since after update there's no difference anymore fullscreen vs borderless.
